migrations/versions/700_add_agreed_variations_to_SupplierFramework.py

```diff
@@ -0,0 +1,22 @@
+"""add agreed_variations to SupplierFramework
+
+Revision ID: 700
+Revises: 690
+Create Date: 2016-08-12 15:52:48.366958
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '700'
+down_revision = '690'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+def upgrade():
+    op.add_column('supplier_frameworks', sa.Column('agreed_variations', postgresql.JSON(), nullable=True))
```

Comment:
So then beyond the race thing, it was realized that of course we want a specific audit event for variation agreement and just relying on `update_supplier_framework_details` and sniffing to see what the caller was _actually_ doing would start making `update_supplier_framework_details` a bit weirder and more complex than it already is, so having a separate endpoint was seen as a good thing. So if we're going to have a separate endpoint it's probably better to not let the fields be manually dickable with too.